### PR TITLE
feat(external docs): build and publish gatsby site when updated

### PR DESCRIPTION
**Pull request by casserni**
Adds a quick action to run on PRs that detects when the /docs directory is updated and then creates a preview of how it affects our Netlify site. Also runs on merge, to deploy the actually changes.

### CODE REVIEW

File: .github/workflows/publish-docs.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+      - name: Setup Node
+        uses: actions/setup-node@v1
+
+      - name: Install Dependencies
+        working-directory: ./vector-website
+        run: |
+          npm config set '//registry.npmjs.org/:_authToken' "${{ secrets.NPM_TOKEN }}"
+          yarn install
+          cd ./plugins/gatsby-plugin-graphql-codegen && yarn install
```
Comment (binarylogic) [inline .github/workflows/publish-docs.yml:56]: @jamtur01 doesn't need to happen now, but should we consider an image for this?

Comment (jamtur01) [reply .github/workflows/publish-docs.yml:56]: How long does it take to `yarn install` etc?

**Review by jamtur01 — APPROVED** [review]
LGTM

File: .github/workflows/publish-docs.yml (hunk 1 above)
Comment (lucperkins) [reply .github/workflows/publish-docs.yml:56]: @jamtur01 About 20 seconds for me locally

**Pull request merged**